src/formatDate.js

```diff
@@ -50,7 +50,62 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newDate = date.split(fromFormat[3]);
+  const toSeparate = toFormat[3];
+  const resultDate = [];
+  let month = '';
+  let day = '';
+  let year = '';
+
+  for (let i = 0; i < newDate.length; i++) {
+    switch (fromFormat[i]) {
+      case 'MM':
+        month = newDate[i];
+        break;
+
+      case 'DD':
+        day = newDate[i];
+        break;
+
+      case 'YY':
+      case 'YYYY':
+        year = newDate[i];
+        break;
+    }
+  }
+
+  for (let i = 0; i < newDate.length; i++) {
+    switch (toFormat[i]) {
+      case 'DD':
+        resultDate[i] = day;
+        break;
+
+      case 'MM':
+        resultDate[i] = month;
+        break;
+
+      case 'YY':
+        if (year.length === 4) {
+          resultDate[i] = year.slice(-2);
+        }
+        break;
+
+      case 'YYYY':
+        if (year.length === 2 && +year < 30) {
+          resultDate[i] = '20' + year;
+          break;
+        }
+
+        if (year.length === 2) {
+          resultDate[i] = '19' + year;
+        } else {
+          resultDate[i] = year;
+        }
+        break;
+    }
```

Comment:
Useless `switch...case`. You'll need only to push `dateObject[toFormat[i]]` value to the resultDate.